src/wp-includes/media.php

```diff
@@ -5614,3 +5834,45 @@ function wp_increase_content_media_count( $amount = 1 ) {
 
 	return $content_media_count;
 }
+
+/**
+ * Add fetchpriority='high' to $loading_attrs if applicable.
+ *
+ * @since 6.3.0
+ * @access private
+ *
+ * @param array[] $loading_attrs Array of the loading attributes for the element.
+ * @param array[] $attr Array of the attributes for the element.
+ * @return array[] $loading_attrs Updated loading attributes for the element.
+ */
+function wp_maybe_add_fetchpriority_high_attr( $loading_attrs, $attr ) {
+	if ( ! empty( $attr['fetchpriority'] ) ) {
+		$loading_attrs['fetchpriority'] = $attr['fetchpriority'];
```

Comment:
See above, we still need to set the flag to `false` to ensure we don't consider other high-priority images if e.g. a developer has already marked it as `high` manually outside of core's logic.
```suggestion
		if ( 'high' === $attr['fetchpriority'] ) {
			$loading_attrs['fetchpriority'] = $attr['fetchpriority'];
			wp_high_priority_element_flag( false );
		}
```

Reply:
Done. Just to note a simple call like get_the_post_thumbnail with `fetchpriority` will also set the highpriority flag. 

```php
get_the_post_thumbnail(
			$post_id,
			'post-thumbnail',
			array(
				'loading'       => false,
				'fetchpriority' => 'high',
			)
);
```